pulp_container/app/viewsets.py

```diff
@@ -114,7 +116,50 @@ class Meta:
         }
 
 
-class TagViewSet(ReadOnlyContentViewSet):
+Repo = namedtuple("Repo", "type perm")
+mirror_repo = Repo("container.container", "container.view_containerrepository")
+push_repo = Repo("container.container-push", "container.view_containerpushrepository")
+
+
+class ContainerQuerySetViewSet(viewsets.GenericViewSet):
+    def get_queryset(self):
+        qs = super().get_queryset()
+        has_model_push_repo = self.request.user.has_perm(push_repo.perm)
+        has_model_repo = self.request.user.has_perm(mirror_repo.perm)
+        if has_model_push_repo and has_model_repo:
+            return qs
+        # if has_model_repo:
+        #    return qs.model.objects.filter(repositories__pulp_type=mirror_repo.type)
+        # if has_model_push_repo:
+        #    return qs.model.objects.filter(repositories__pulp_type=push_repo.type)
+        query_params = self.request.query_params
+        if query_params and "repository_version" in query_params.keys():
+            policy = self.permission_classes[0]
+            repos_can_see, push_repos_can_see = policy.has_repo_query_params(
```

Comment:
At the very least, the name of this function confuses me. It sounds like it's supposed to return bool.